### ISSUE TITLE
Allow unordered keys for substituion variables

### ISSUE THREAD

**Issue by DDtKey** [dotenv-linter/dotenv-linter#382]
In variable substitution cases, `dotenv-linter` shouldn't complain on order of the keys.

Such files to be considered valid:

```env
# .env
FOO=something
BAR=$FOO # this is correct, because BAR uses FOO variable
```
For this, changes must be made to the` UnorderedKeyChecker`.

Since we already support sort groups (see the [Unordered Key documentation](https://dotenv-linter.github.io/#/checks/unordered_key)), a good way would be just consider the substitution key (which refers to another key in [current sort group](https://github.com/dotenv-linter/dotenv-linter/blob/492d3a4046fd97cdfbeb7484e430375683a06a2f/src/checks/unordered_key.rs#L6)) - as another group separator.

### TODO
Using the substitution key getter (see #379), check if this key refers to another key in the [current sort group](https://github.com/dotenv-linter/dotenv-linter/blob/492d3a4046fd97cdfbeb7484e430375683a06a2f/src/checks/unordered_key.rs#L6), and then start a new sort group [here](https://github.com/dotenv-linter/dotenv-linter/blob/492d3a4046fd97cdfbeb7484e430375683a06a2f/src/checks/unordered_key.rs#L31) if so. 

### Example

This file is incorrect because the keys are not sorted alphabetically before and after the `FOO` key (which uses `KEY`):

```env
# .env
KEY=VALUE
ABC=XYZ
FOO=$KEY
BOO=$FOO
XYZ=ABC
BAR=FOO
```

The correct version of this file should look like this:

```env
# .env
ABC=XYZ
KEY=VALUE
FOO=$KEY # Unordered, because FOO uses KEY
BAR=FOO # here starts new sort group
BOO=$FOO
XYZ=ABC
```
